osu.Game/Overlays/SkinEditor/SkinEditor.cs

```diff
@@ -361,6 +421,51 @@ private void revert()
             }
         }
 
+        protected void Cut()
+        {
+            Copy();
+            DeleteItems(SelectedComponents.ToArray());
+        }
+
+        protected void Copy()
+        {
+            clipboard.Content.Value = JsonConvert.SerializeObject(SelectedComponents.Cast<Drawable>().Select(s => s.CreateSerialisedInfo()).ToArray());
+        }
+
+        protected void Clone()
+        {
+            // Avoid attempting to clone if copying is not available (as it may result in pasting something unexpected).
+            if (!canCopy.Value)
+                return;
+
+            Copy();
+            Paste();
+        }
+
+        protected void Paste()
+        {
+            changeHandler?.BeginChange();
+
+            var drawableInfo = JsonConvert.DeserializeObject<SerialisedDrawableInfo[]>(clipboard.Content.Value);
+
+            if (drawableInfo == null)
+                return;
+
+            var instances = drawableInfo.Select(d => d.CreateInstance())
+                                        .OfType<ISerialisableDrawable>()
+                                        .ToArray();
+
+            SelectedComponents.Clear();
+
+            foreach (var i in instances)
+            {
+                if (placeComponent(i, false))
+                    SelectedComponents.Add(i);
```

Comment:
`placeComponent()` already adds the placed component to selected:

https://github.com/ppy/osu/blob/43d33d45caa852707a1b5064f8cb23549ddc45de/osu.Game/Overlays/SkinEditor/SkinEditor.cs#L389-L390

so this ends up double-adding the pasted component if copying single. This is noticeable by the number of selected items being wrong in the top left corner of the selection box:

https://user-images.githubusercontent.com/20418176/220181487-06a0003a-718c-4451-815e-c2a39e632c74.mp4

but it also breaks pasting multiple items (only the last pasted item will be selected).

I would probably fix this by removing the selection logic from `placeComponent()` and moving it locally to the two other calls that used to depend on it rather than add another boolean flag argument to `placeComponent()`, but not entirely sure if you'd agree with that.

Reply:
Would you be against only removing the `Clear` call from `placeComponent` and having that called at the call site? I'm not sure if this reads better or not, but my thinking is that we have zero scenarios where `placeComponent` it called but the new placement *isn't* added to selection, so that part should still be coupled.

Reply:
Sure, I can agree with moving the clear only.